crates/bevy_ecs/src/query/state.rs

```diff
@@ -187,6 +187,24 @@ impl<D: QueryData, F: QueryFilter> QueryState<D, F> {
             }
         }
         state.archetype_generation = world.archetypes.generation();
+
+        // Resource access is not part of any archetype and must be handled separately
+        if state.component_access.access().has_read_all_resources() {
+            access.read_all_resources();
+        } else {
+            for component_id in state.component_access.access().resource_reads() {
+                access.add_resource_read(world.initialize_resource_internal(component_id).id());
+            }
+        }
+
+        if state.component_access.access().has_write_all_resources() {
```

Comment:
This will also support *mutable* resource access in queries, mainly because it was straightforward to do so, and because allowing `access.add_resource_write` to compile but do nothing seemed dangerous.  But it's hard to do anything useful with mutable resource access.  Multiple query items may be alive at once, so you can't use a mutable borrow in the fetch item.  

Would it make more sense to simply prohibit mutable access to resources, and panic here if there is any?